versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/RocksBaseValue.java

```diff
@@ -164,6 +175,259 @@ protected static String conditionNotMatchedMessage(Function function) {
       function.getRootPathAsNameSegment().getName());
   }
 
+  /**
+   * Entry point to update this object with a series of UpdateClauses.
+   * @param updates the updates to apply
+   */
+  @Override
+  public void update(UpdateExpression updates) {
+    List<ExpressionPath.PathSegment> removedPaths = removeUnecessaryFunctions(updates.getClauses())
+        .stream()
+        .map(this::updateWithFunction)
+        .filter(Objects::nonNull)
+        .sorted(new Comparator<ExpressionPath.PathSegment>() {
+          /* Sort the PathSegments as follows
+           * names before positions
+           * names by string order
+           * higher positions before lower positions
+           * path with child before path without child
+           * when equal, recurse in with the child segments
+           */
+          @Override
+          public int compare(ExpressionPath.PathSegment ps1, ExpressionPath.PathSegment ps2) {
+            if (ps1.isName() && ps2.isName()) {
```

Comment:
Is this comparison logic specific to this application, or should it be within the PathSegment?

Reply:
This is specific to applying updates. The paths are sorted here into an order that makes them safe to apply. I don't think that this would be generally useful to the PathSegment class.